forge/src/multi_runner.rs

```diff
@@ -55,34 +56,76 @@ impl MultiContractRunnerBuilder {
         // This is just the contracts compiled, but we need to merge this with the read cached
         // artifacts
         let contracts = output.into_artifacts();
-        let contracts: BTreeMap<String, (Abi, Address, Vec<String>)> = contracts
-            // only take contracts with valid abi and bytecode
-            .filter_map(|(fname, contract)| {
-                let (abi, bytecode) = contract.into_inner();
-                abi.and_then(|abi| bytecode.map(|bytecode| (fname, abi, bytecode)))
-            })
-            // Only take contracts with empty constructors.
-            .filter(|(_, abi, _)| {
-                abi.constructor.as_ref().map(|c| c.inputs.is_empty()).unwrap_or(true)
-            })
-            // Only take contracts which contain a `test` function
-            .filter(|(_, abi, _)| abi.functions().any(|func| func.name.starts_with("test")))
-            // deploy the contracts
-            .map(|(name, abi, bytecode)| {
-                let span = tracing::trace_span!("deploying", ?name);
-                let _enter = span.enter();
-
-                let (addr, _, _, logs) = evm
-                    .deploy(sender, bytecode, 0.into())
-                    .wrap_err(format!("could not deploy {}", name))?;
-
-                evm.set_balance(addr, initial_balance);
-                Ok((name, (abi, addr, logs)))
-            })
-            .collect::<Result<BTreeMap<_, _>>>()?;
+        let mut known_contracts: BTreeMap<String, (Abi, Vec<u8>)> = Default::default();
+        let mut deployed_contracts: BTreeMap<String, (Abi, Address, Vec<String>)> =
+            Default::default();
+
+        use std::any::Any;
+        for (fname, contract) in contracts {
```

Comment:
tbh this probably isnt correct with the changes made on master and our dependencies. but all tests pass